packages/ui/src/components/ComponentList/index.tsx

```diff
@@ -5,60 +5,167 @@ import styled from "styled-components";
 import {Component} from "../../models";
 import {paddings} from "../../styles";
 
-const StyledComponentList = styled.div`
-  display: flex;
-  flex-direction: column;
+const StyledComponentList = styled.div``;
+
+const StyledFileTree = styled.ul`
+  margin: 0;
+  padding: 0;
+  list-style: none;
 `;
 
-const ComponentName = styled.div<{selected?: boolean}>`
+const StyledLink = styled(Link)`
   display: inline-block;
   overflow-x: hidden;
   text-overflow: ellipsis;
   text-decoration: none;
-  padding: ${paddings.small} ${paddings.medium};
-  ${props => props.selected && `background-color: ${props.theme.colors.bg}`}
-  font-size: ${props => props.theme.fontSizes.subtitle};
   color: ${props => props.theme.colors.text};
-
   &:hover {
-    color: ${props => props.theme.colors.textSecondary};
+    color: ${props => props.theme.colors.textTertiary};
   }
 `;
 
-const Path = styled.span`
-  font-style: italic;
-  font-weight: lighter;
+interface FileProps {
+  level: number;
+  selected?: boolean;
+}
+
+const StyledDirectory = styled.span<FileProps>`
+  display: block;
+  padding: ${paddings.small};
+  padding-left: calc(${paddings.medium} * (${props => props.level} + 1));
```

Comment:
I think the indent is a little large. In VSCode it is very small

<img width="226" alt="image" src="https://user-images.githubusercontent.com/3044853/58248981-8d6e0280-7d55-11e9-992d-d5ae19ecf113.png">

Compare to

![image](https://user-images.githubusercontent.com/3044853/58248996-98c12e00-7d55-11e9-84ef-b41c5cd17d74.png)

With a large file tree there might be a lot of horizontal scroll